roles/otel_collector/vars/main.yml

```diff
@@ -0,0 +1,2 @@
+otel_collector_binary_url: "https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/v{{ otel_collector_version }}/{% if otel_collector_type == 'contrib' %}otelcol-contrib_{{ otel_collector_version }}_linux_{{ otel_collector_arch }}{% else %}otelcol_{{ otel_collector_version }}_linux_{{ otel_collector_arch }}{% endif %}.tar.gz"
```

Comment:
Perhaps move this from `vars/` to `defaults/` so that it can be overwritten? Allows you to use a self hosted mirror of opentelemetry collector which is needed on air-gapped networks.

Reply:
Got it, Yeah makes sense. Though this does use `otel_collector_arch` which is present in install.yml, Woudn't placing this in defaults have issues at runtime?

Reply:
I think the variable evaluated on runtime so it should work, but ansible-lint could complain. But I don't see why you shouldn't be able to move `otel_collector_arch` to `vars/` instead of using `set_fact`.